frontend/src/plugins/core/registerReactComponent.tsx

```diff
@@ -144,13 +118,15 @@ function PluginSlotInternal<T>(
     };
   }, [hostElement, plugin.validator]);
 
-  const setValueAndSendInput = useCallback(
-    (value: T | ((prevValue: T) => T)): void => {
-      shouldDispatchInput.current = true;
-      setValue(value);
-    },
-    [setValue, shouldDispatchInput]
-  );
+  // When the value changes, send an input event
+  const setValueAndSendInput = useEvent((value: SetStateAction<T>): void => {
+    setValue((prevValue) => {
+      const updater = Functions.asUpdater(value);
+      const nextValue = updater(prevValue);
+      hostElement.dispatchEvent(createInputEvent(nextValue, hostElement));
+      return nextValue;
```

Comment:
by dispatching here (and instead of the next render loop), we are able to resubmit since React otherwise skips re-renders when `setValue` is `===` to the previous value